src/datasets/load.py

```diff
@@ -803,10 +811,14 @@ def __init__(
         self,
         name: str,
         revision: Optional[Union[str, Version]] = None,
+        data_dir: Optional[str] = None,
         data_files: Optional[Union[str, List, Dict]] = None,
         download_config: Optional[DownloadConfig] = None,
         download_mode: Optional[DownloadMode] = None,
     ):
+        if data_files is None and data_dir is not None:
+            data_files = [os.path.join(data_dir, "*"), os.path.join(data_dir, "**/*")]
```

Comment:
This would put everything in the train split no ? This goes against https://huggingface.co/docs/datasets/repository_structure.html#multiple-configuration-wip

We can work on this later in another PR

Reply:
I don't think it "goes against" that as long as the user doesn't specify `data_dir`, and if he does, then it's safe to assume that we can turn off the repository structure inference 🙂 .

Reply:
Ok sounds good, this will require some documentation then